cpp/src/arrow/io/interfaces.h

```diff
@@ -49,16 +49,36 @@ struct ReadRange {
 };
 
 // EXPERIMENTAL
```

Comment:
```suggestion
/// EXPERIMENTAL
/// Includes an Executor in (which will be used to execute asynchronous reads),
/// a MemoryPool (which will be used to allocate buffers when zero copy reads
/// are not possible), and an external id (in case the executor receives tasks from
/// multiple sources and must distinguish tasks associated with this IOContext).
```